### PR TITLE
feat: convert groundtest to chisel3

### PR DESCRIPTION
**Pull request by tianrui-wei**
Convert groundtest subdirectory to chisel3

Signed-off-by: Tianrui Wei <tianrui@tianruiwei.com>

**Type of change**: other enhancement

**Impact**: no functional change

**Development Phase**: implementation

**Release Notes**

### CODE REVIEW

**Review by sequencer — CHANGES_REQUESTED** [review]
small nitpicks

File: src/main/scala/groundtest/TraceGen.scala [hunk 1]
```diff
@@ -210,33 +211,33 @@ class TagMan(val logNumTags : Int) extends Module {
...
-  reqTimer.io.start.valid := io.mem.req.fire()
+  reqTimer.io.start.valid := io.mem.req.fire
   reqTimer.io.start.bits := io.mem.req.bits.tag
   reqTimer.io.stop.valid := io.mem.resp.valid
   reqTimer.io.stop.bits := io.mem.resp.bits.tag
 
   // Random addresses
   // ----------------
-  
+
```
[the remark this answers is not in the document]
Comment (tianrui-wei) [reply src/main/scala/groundtest/TraceGen.scala:233]: addressed in the new commit

**Comment by tianrui-wei** [comment]
@sequencer Thank you for reviewing, the issues about is addressed

File: src/main/scala/groundtest/DummyPTW.scala [hunk 2]
```diff
@@ -3,7 +3,8 @@
 
 package freechips.rocketchip.groundtest
 
-import Chisel._
+import chisel3._
+import chisel3.util._
```
[the remark this answers is not in the document]
Comment (tianrui-wei) [reply src/main/scala/groundtest/DummyPTW.scala]: Addressed in the new commit

File: src/main/scala/groundtest/DummyPTW.scala [hunk 3]
```diff
@@ -12,44 +13,44 @@ import freechips.rocketchip.util.ParameterizedBundle
 
 class DummyPTW(n: Int)(implicit p: Parameters) extends CoreModule()(p) {
   val io = new Bundle {
```
[the remark this answers is not in the document]
Comment (tianrui-wei) [reply src/main/scala/groundtest/DummyPTW.scala]: Addressed in the new commit

File: src/main/scala/groundtest/DummyPTW.scala [hunk 4]
```diff
@@ -12,44 +13,44 @@ import freechips.rocketchip.util.ParameterizedBundle
...
-    val requestors = Vec(n, new TLBPTWIO).flip
+    val requestors = Flipped(Vec(n, new TLBPTWIO))
   }
```
[the remark this answers is not in the document]
Comment (tianrui-wei) [reply src/main/scala/groundtest/DummyPTW.scala]: Addressed in the new commit

File: src/main/scala/groundtest/TraceGen.scala [hunk 5]
```diff
@@ -19,7 +19,8 @@
 
 package freechips.rocketchip.groundtest
  
-import Chisel._
+import chisel3._
+import chisel3.util._
```
[the remark this answers is not in the document]
Comment (tianrui-wei) [reply src/main/scala/groundtest/TraceGen.scala]: Addressed in the new commit

File: src/main/scala/groundtest/TraceGen.scala [hunk 6]
```diff
@@ -210,33 +211,33 @@ class TagMan(val logNumTags : Int) extends Module {
 class TraceGenerator(val params: TraceGenParams)(implicit val p: Parameters) extends Module
     with HasTraceGenParams {
   val io = new Bundle {
```
[the remark this answers is not in the document]
Comment (tianrui-wei) [reply src/main/scala/groundtest/TraceGen.scala]: Addressed in the new commit

File: src/main/scala/groundtest/TraceGen.scala [hunk 7]
```diff
@@ -566,21 +567,21 @@ class TraceGenerator(val params: TraceGenParams)(implicit val p: Parameters) ext
...
   // Termination condition
   // ---------------------
 
-  val done = reqCount  === UInt(numReqsPerGen) &&
-             respCount === UInt(numReqsPerGen)
+  val done = reqCount  === numReqsPerGen.U &&
+             respCount === numReqsPerGen.U
 
-  val donePulse = done && !Reg(init = Bool(false), next = done)
+  val donePulse = done && !RegNext(done, init=false.B)
```
Comment (sequencer) [inline src/main/scala/groundtest/TraceGen.scala:584]: Chisel 3.6 won't be happy with this named param
```suggestion
  val donePulse = done && !RegNext(done, false.B)
```

**Review by sequencer — CHANGES_REQUESTED** [review]

File: src/main/scala/groundtest/TraceGen.scala (hunk 7 above)
Comment (tianrui-wei) [reply src/main/scala/groundtest/TraceGen.scala]: Thanks for the tip

**Comment by tianrui-wei** [comment]
Thanks for fixing the named param. The CI seems all good after your commit

**Review by sequencer — APPROVED** [review]

**Pull request merged**